WasatchNET/TetonSpectrometer.cs

```diff
@@ -0,0 +1,517 @@
+        {
+            byte[] wrapped = new byte[payload.Length + 6 + padding];
+            uint cmdLength = (ushort)(1 + payload.Length);
+            uint len1 = cmdLength & 0xFF;
+            uint len0 = (cmdLength >> 8) & 0xFF;
+            wrapped[0] = START_CMD;
+            wrapped[1] = (byte)len0;
+            wrapped[2] = (byte)len1;
+            wrapped[3] = command;
+            int index = 4;
+            for (int i = 0; i < payload.Length; ++i)
+            {
+                wrapped[i + 4] = payload[i];
+                ++index;
+            }
+
+            byte[] crcLoad = new byte[payload.Length + 3];
+
+            for (int i = 0; i < payload.Length + 3; ++i)
+            {
+                crcLoad[i] = wrapped[i + 1];
+            }
+
+            wrapped[index++] = Calc_CRC_8(crcLoad, crcLoad.Length);
+            wrapped[index] = END_CMD;
+
+            return wrapped;
+        }
+
+        static byte[] padding(int size)
+        {
+            byte[] padding = new byte[size];
+            for (int i = 0; i < size; ++i)
+                padding[i] = 0x00;
+
+            return padding;
+        }
+
+        override internal bool open()
+        {
+            eeprom = new EEPROM(this);
+
+            int errorReader = 0;
+
+            string ftdi = FtdiInventory.DeviceListInfo();
+            if (ftdi.Length == 0)
+            {
+                logger.info("Unable to find any Teton spectrometer, try again");
+                return false;
+            }
+            else
+            {
+                string snPattern = "Serial Number: ";
+
+                int start = ftdi.IndexOf(snPattern, 0);
+
+                start = start + 15;
+
+                string devSerialNumber = ftdi.Substring(start, 8);
+
+                MpsseDevice.MpsseParams mpsseParams = new MpsseDevice.MpsseParams();
+                mpsseParams.clockDevisor = 1;
+
+                try
+                {
+                    mpsse = new FT232H(devSerialNumber, mpsseParams);
+                }
+                catch (Exception e)
+                {
+                    logger.info("Unable to create MPSSE connection with board. May be missing drivers");
+                    return false;
+                }
+
+                try
+                {
+                    spi = new SpiDevice(mpsse,
+                             new SpiDevice.SpiParams
+                             {
+                                 Mode = SpiDevice.SpiMode.Mode0,
+                                 ChipSelect = FtdiPin.CS,
+                                 ChipSelectPolicy = SpiDevice.CsPolicy.CsActiveLow
+                             });
+                }
+                catch (Exception e)
+                {
+                    logger.info("Unable to create SPI connection with board. May be missing drivers");
+                    return false;
+                }
+
+                pixels = 1024;
```

Comment:
This *will* come back to bite you someday :-)

Reply:
I agree I really need to keep an eye on this and not let it stay this way. However it's there because the "get pixel count" command was giving me issues.

Reply:
I like to at least write "// kludge" or "@todo" in there so someday someone will realize it needs to be fixed.  For now it's certainly okay to state "all supported SPI spectrometers have 1024 pixels, and the API does not yet provide an instruction for specifying pixel count".